Model/Api/AdyenDonationCampaigns.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+namespace Adyen\Payment\Model\Api;
+
+use Adyen\Payment\Api\AdyenDonationCampaignsInterface;
+use Adyen\Payment\Helper\ChargedCurrency;
+use Adyen\Payment\Helper\DonationsHelper;
+use Magento\Framework\Exception\LocalizedException;
+use Adyen\Payment\Model\Sales\OrderRepository;
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\Sales\Api\Data\OrderInterface;
+
+class AdyenDonationCampaigns implements AdyenDonationCampaignsInterface
+{
+    private DonationsHelper $donationsHelper;
+    private OrderRepository $orderRepository;
+    private ChargedCurrency $chargedCurrency;
+
+    public function __construct(
+        DonationsHelper $donationsHelper,
+        OrderRepository $orderRepository,
+        ChargedCurrency $chargedCurrency
+    ) {
+        $this->donationsHelper = $donationsHelper;
+        $this->orderRepository = $orderRepository;
+        $this->chargedCurrency = $chargedCurrency;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+
+    public function getCampaigns(int $orderId, string $payload): string
+    {
+        $order = $this->orderRepository->get($orderId);
+
+        if (!$order->getEntityId()) {
+            throw new LocalizedException(__('Order not found.'));
```

Comment:
`LocalizedException` shouldn't give details about error details. This can be misused by the malicious actors. The a generic exception message can be used (ideally identical to the message we use in DonationHelper class for LocalizedException - ie: `Donation failed!`). Besides that, it might also throw EntityNotFound (or similar) exception if the user is not authorized to fetch the given order.